packages/ember-metal/lib/property_set.js

```diff
@@ -77,7 +77,7 @@ export function set(obj, keyName, value, tolerant) {
   } else if (currentValue === undefined && 'object' === typeof obj && !(keyName in obj) &&
     typeof obj.setUnknownProperty === 'function') { /* unknown property */
     obj.setUnknownProperty(keyName, value);
-  } else if (currentValue === value) { /* no change */
+  } else if (currentValue === value && keyName in obj) { /* no change */
```

Comment:
I feel like removing this line too, thoughts ?

Reply:
Agree...